crates/cadenza-gcode/src/syntax.rs

```diff
@@ -179,44 +184,138 @@ impl<'src> Parser<'src> {
     }
 
     fn parse_parameter(&mut self) {
-        // Parameter is like X100 or Y50 or just X (flag)
-        // We'll parse it as: [X, 100] where X is the receiver (letter) and 100 is the argument
+        // Parameter can be:
+        // 1. X100 or Y50 (traditional GCode - single letter + value)
+        // 2. PIN=my_led (Klipper named parameter - multi-char identifier with =)
+        // 3. X (flag without value)
 
         let letter_start = self.pos;
+
+        // Parse the first character (letter)
         self.pos += 1;
+
+        // Check if this is a Klipper-style named parameter (more letters/underscores before =)
+        let mut is_klipper_style = false;
+        let save_pos = self.pos;
+
+        while self.pos < self.src.len() {
+            let ch = self.peek_char().unwrap();
+            if ch.is_ascii_alphabetic() || ch == '_' {
+                self.pos += 1;
+                is_klipper_style = true;
+            } else if ch == '=' && is_klipper_style {
+                // This is Klipper format
+                break;
+            } else {
+                // Not Klipper format, restore position
+                self.pos = save_pos;
+                break;
+            }
+        }
+
         let letter_text = &self.src[letter_start..self.pos];
 
-        // Check if there's a value after the letter
-        let has_value = self.pos < self.src.len()
-            && self
-                .peek_char()
-                .is_some_and(|c| c.is_ascii_digit() || c == '.' || c == '-');
+        // Check for equals sign (Klipper format: NAME=value)
+        if self.peek_char() == Some('=') {
+            self.pos += 1; // Skip the '='
```

Comment:
If we skip the = then not every byte is accounted for in the CST. Speaking of, we need to add a test for that now, since it's come up a lot. We should iterate over every node in the cst and assert the spans cover the whole file and that the span actually indexes into the exact same text as the original source.

But I think for these types of params, we should pass the equals as an apply: `[=, key, value]` so the handler macro can use that information.